### PR TITLE
item charges: update amulet of chemistry charges when creating less than 4-dose potions

### LINKED ISSUE #11822: Amulet of Chemistry charges don't update when making potions with less than 4 doses.
Currently, the regex that checks whether the amulet of chemistry has lost a charge assumes that the player is making 4 dose potions. 
The amulet works on all potions however, and players often create stamina potions out of Super Energy (1) or (2) to maximise the extra doses.

It would be a very simple change to have the regex support any dose of potion being made.

Thanks to `Blitzburghbrian` on Discord for bringing this up.

### PR DESCRIPTION
**Pull request by MMagicala**
Closes #11822

Updated the regex for amulet of chemistry to update the amulet's charge count when creating less than 4-dose potions

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 1]
```diff
@@ -84,9 +84,11 @@
 		"Your amulet of chemistry has (\\d) charges? left\\."
 	);
 	private static final Pattern AMULET_OF_CHEMISTRY_USED_PATTERN = Pattern.compile(
-		"Your amulet of chemistry helps you create a 4-dose potion\\. (?:<col=ff0000>)?It has (\\d|one) charges? left\\."
+		"Your amulet of chemistry helps you create a (\\d)-dose potion\\. (?:<col=ff0000>)?It has (\\d|one) charges? left\\."
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java:87]:
```suggestion
		"Your amulet of chemistry helps you create a \\d-dose potion\\. (?:<col=ff0000>)?It has (\\d|one) charges? left\\."

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 2]
```diff
@@ -84,9 +84,11 @@
...
+	);
+	private static final Pattern AMULET_OF_CHEMISTRY_BREAK_PATTERN = Pattern.compile(
+		"Your amulet of chemistry helps you create a (\\d)-dose potion\\. (?:<col=ff0000>)?It then crumbles to dust\\."
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java:90]:
```suggestion
			"Your amulet of chemistry helps you create a \\d-dose potion\\. (?:<col=ff0000>)?It then crumbles to dust\\."
```

**Comment by Nightfirecat** [comment]
Please add a test for this case.

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 3]
```diff
@@ -84,9 +84,11 @@
 		"Your amulet of chemistry has (\\d) charges? left\\."
 	);
 	private static final Pattern AMULET_OF_CHEMISTRY_USED_PATTERN = Pattern.compile(
-		"Your amulet of chemistry helps you create a 4-dose potion\\. (?:<col=ff0000>)?It has (\\d|one) charges? left\\."
+		"Your amulet of chemistry helps you create a \\d-dose potion\\. (?:<col=ff0000>)?It has (\\d|one) charges? left\\."
```
Comment (MMagicala) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java:87]: To test the number of doses, don't we have to use a capture group? The plugin only captures the number of charges.
```suggestion
		"Your amulet of chemistry helps you create a (\\d)-dose potion\\. (?:<col=ff0000>)?It has (\\d|one) charges? left\\."
```

**Pull request merged**